wrapper/src/main/java/software/amazon/jdbc/plugin/DefaultConnectionPlugin.java

```diff
@@ -231,7 +256,10 @@ public HostSpec getHostSpecByStrategy(HostRole role, String strategy)
       throw new SQLException(Messages.get("DefaultConnectionPlugin.noHostsAvailable"));
     }
 
-    return this.connProviderManager.getHostSpecByStrategy(hosts, role, strategy, this.pluginService.getProperties());
+    if (this.effectiveConnProvider != null) {
+      return this.effectiveConnProvider.getHostSpecByStrategy(hosts, role, strategy);
```

Comment:
return this.effectiveConnProvider.getHostSpecByStrategy(hosts, role, strategy, this.pluginService.getProperties());